socless/integrations.py

```diff
@@ -34,111 +34,98 @@ class ParameterResolver:
+            resolved_list = []
+            for item in reference_path:
+                resolved_list.append(self.resolve_reference(item))
+            return resolved_list
         else:
             return reference_path
 
-        if conversion:
-            resolved = self.apply_conversion_from(resolved, conversion)
-        return resolved
-
     def resolve_parameters(self, parameters):
         """Resolve a set of parameter references to their actual vaules
-
         Args:
             parameters (dict): Parameter references to resolve
         Returns:
             a dictionary containing resolved parameter references
         """
         actual_params = {}
-        for parameter, reference in list(parameters.items()):
+        for parameter, reference in parameters.items():
             actual_params[parameter] = self.resolve_reference(reference)
         return actual_params
 
-    def apply_conversion_from(self, data, conversion):
-        """Convert the data type of a parameter
 
-        Handles conversion of the datatype of a parameter intended for an integration
+def convert_deprecated_vault_to_template(vault_reference) -> str:
+    reference, _, conversion = vault_reference.partition(CONVERSION_TOKEN)
+    _, _, file_id = reference.partition(VAULT_TOKEN)
+    template = f"vault('{file_id}')"
+    if conversion:
+        template = template + " |fromjson"
+    return "{" + template + "}"
 
-        Args:
-            data (str): The data to convert
-            conversion (str): The conversion to apply
-        Returns:
-            str: The converted data
-        """
-        if conversion == "json":
-            return json.loads(data)
+
+def convert_legacy_reference_to_template(reference_path: str) -> str:
+    """Allow backwards compatibility for legacy socless parameter reference to jinja templates.
+    `render_jinja_template` jinja translation is supported by this function that converts
+    legacy references to valid jinja templates according to the table below:
+
+        Legacy Reference   | Converted Jinja template
+        ----------------   | ------------------------
+        “$.<ref_path>”     |  “{context.<ref_path>}”
+        “vault:<vault-id>” |  “vault-id | fromvault}”
+        "<something>!json" |  "{<something> | fromjson}"
+    """
+    try:
+        template = reference_path
+
+        if template.startswith(PATH_TOKEN):
+            _, _, conversion = template.partition(CONVERSION_TOKEN)
+            template = f"context{template[1:]}"
+            if conversion:
+                template = template + " |fromjson"
+            template = "{" + template + "}"
+        elif template.startswith(VAULT_TOKEN):
+            template = convert_deprecated_vault_to_template(template)
+
+        return template
+    except (TypeError, KeyError) as e:
+        raise SoclessException(
+            f"Unable to convert reference type {type(reference_path)} to template - {e}"
+        )
```

Comment:
We've got a good deal of repeated code between these two functions. What's the driver for that decision?

Reply:
moved the repeated lines into a descriptive function